src/state/routing/types.ts

```diff
@@ -62,3 +65,32 @@ export interface GetQuoteResult {
   route: Array<V3PoolInRoute[] | V2PoolInRoute[]>
   routeString: string
 }
+
+export class InterfaceTrade<
```

Comment:
after talking with pote, I think we prefer to use composition over inheritance here, meaning

```
export type TradeWithGasEstimates = {
  trade: Trade<>,
  gasEstimates: {
    gasUseEstimateUSD,
  }
}
```

or something like that. WDYT?

Reply:
First off, like this name better. 

Yeah im okay using composition - probably makes it easier if `Trade` is updated in future